File: libsignal-service/src/main/java/org/whispersystems/signalservice/api/subscriptions/PayPalCreatePaymentIntentResponse.java
package org.whispersystems.signalservice.api.subscriptions;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Response object from creating a payment intent via PayPal
 */
public class PayPalCreatePaymentIntentResponse {

  private final String approvalUrl;
  private final String paymentId;

  @JsonCreator
  public PayPalCreatePaymentIntentResponse(@JsonProperty("approvalUrl") String approvalUrl, @JsonProperty("paymentId") String paymentId) {
    this.approvalUrl = approvalUrl;
    this.paymentId   = paymentId;
  }

  public String getApprovalUrl() {
    return approvalUrl;
  }

  public String getPaymentId() {
    return paymentId;
  }
}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/api/subscriptions/PayPalCreatePaymentIntentResponse.kt
/*
 * Copyright 2025 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.api.subscriptions

/**
 * Response object from creating a payment intent via PayPal
 */
data class PayPalCreatePaymentIntentResponse(
  val approvalUrl: String,
  val paymentId: String
)
